### PR TITLE
95 property singularity determination

### PR DESCRIPTION
**Pull request by alishaevn**
# this is a breaking change

# todo
- test uploading this yml file: [utk_final.yml.zip](https://github.com/samvera-labs/allinson_flex/files/9428581/utk_final.yml.zip)
- import this csv: [all_metadata.csv](https://github.com/samvera-labs/allinson_flex/files/9428582/all_metadata.csv)
- make sure that properties in the yml file that have `cardinality: maximum: #` are limited to only having that amount of values

### CODE REVIEW

File: app/forms/concerns/allinson_flex/dynamic_form_behavior.rb [hunk 1]
```diff
@@ -15,9 +15,9 @@ module DynamicFormBehavior
                          :visibility_during_embargo, :embargo_release_date, :visibility_after_embargo,
                          :visibility_during_lease, :lease_expiration_date, :visibility_after_lease,
                          :visibility, :ordered_member_ids, :source, :in_works_ids,
-                         :member_of_collection_ids, :admin_set_id, :profile_version]
```
Comment (alishaevn) [inline app/forms/concerns/allinson_flex/dynamic_form_behavior.rb:18]: `profile_version` is already on line 14.

File: app/models/concerns/allinson_flex/dynamic_solr_document.rb [hunk 2]
```diff
@@ -25,7 +25,8 @@ def load_allinson_flex
               attribute(
                 prop.name,
                 # if the property is singular, make it so
-                prop.cardinality_maximum == 1 ? Hyrax::SolrDocument::Metadata::Solr::String : Hyrax::SolrDocument::Metadata::Solr::Array,
+                # WIP(alishaevn): don't base this on cardinality
+                prop.multi_value ? Hyrax::SolrDocument::Metadata::Solr::Array : Hyrax::SolrDocument::Metadata::Solr::String,
```
Comment (alishaevn) [inline app/models/concerns/allinson_flex/dynamic_solr_document.rb:29]: I'm not actually sure what should be an array and what should be a string by default. don't think this is defined anywhere in samvera. so I made the spreadsheet below from scratch and intended to import it and basically wait until hyrax told me that it expected one thing but I gave it another.

not the most efficient. but I'm unsure how else to go about it.
[all_metadata.csv](https://github.com/samvera-labs/allinson_flex/files/9428367/all_metadata.csv)

File: lib/generators/allinson_flex/templates/config/metadata_profile/default_schema.json [hunk 3]
```diff
@@ -422,6 +422,14 @@
...
             ]
           },
+          "multi_value": {
+            "type": "boolean",
+            "description": "Set whether the property is singular or multiple",
+            "default": "true",
+            "examples": [
+              true
+            ]
+          },
```
Comment (alishaevn) [inline lib/generators/allinson_flex/templates/config/metadata_profile/default_schema.json:432]: this is just how I imagine things will go. but....we can decide that this defaults to false instead too.

**Comment by ShanaLMoore** [comment]
work continued on with https://github.com/samvera-labs/allinson_flex/pull/97/files

**Pull request closed without merging**